### ISSUE TITLE
UI: cannot create a workspace from default Linux AWS template

### ISSUE THREAD

**Issue by nadzeyav** [coder/coder#1977]
## OS Information

- OS: Windows
- Browser (if applicable): Chrome, Firefox
- Architecture:
- `coder --version`:
[Coder v0.0.0-devel+9e3a625](https://github.com/coder/coder/commit/9e3a62589898119b38fdfd52a92380bada45fa4b)

## Steps to Reproduce

1. Create a template Linux on AWS EC2 using build it templates
2. Login to coder
3. Click Create workspace
4. Choose aws-linux template

## Expected

Options to choose region are displayed

## Actual

Error is displayed

## Screenshot

![image](https://user-images.githubusercontent.com/9592232/171629244-e9fce40e-07da-4204-9b0e-a5877f236b27.png)

## Notes

======================= STACK TRACE ========================

Cannot read properties of null (reading 'length')
    at J (https://vigorous-germain0.try.coder.app/bundle.595fd896d2995adf1ea9.js:1:7186)
    at aa (https://vigorous-germain0.try.coder.app/bundle.f605f7e1baaad8f0b827.js:2:560721)
    at Hs (https://vigorous-germain0.try.coder.app/bundle.f605f7e1baaad8f0b827.js:2:612770)
    at Au (https://vigorous-germain0.try.coder.app/bundle.f605f7e1baaad8f0b827.js:2:600044)
    at ku (https://vigorous-germain0.try.coder.app/bundle.f605f7e1baaad8f0b827.js:2:599972)
    at Cu (https://vigorous-germain0.try.coder.app/bundle.f605f7e1baaad8f0b827.js:2:599835)
    at gu (https://vigorous-germain0.try.coder.app/bundle.f605f7e1baaad8f0b827.js:2:596822)
    at https://vigorous-germain0.try.coder.app/bundle.f605f7e1baaad8f0b827.js:2:546643
    at t.unstable_runWithPriority (https://vigorous-germain0.try.coder.app/bundle.f605f7e1baaad8f0b827.js:2:647788)
    at Hi (https://vigorous-germain0.try.coder.app/bundle.f605f7e1baaad8f0b827.js:2:546420)

============================================================

**Comment by kconley-sq** [comment]
I believe I ran into this issue myself in `v0.6.0`:

```
======================= STACK TRACE ========================

Cannot read properties of null (reading 'length')
    at Y (https://coder.sqcorp.co/bundle.d348b2254471bdd9d116.js:1:7754)
    at aa (https://coder.sqcorp.co/bundle.82f6fac05ecb071bdc27.js:2:333661)
    at Hs (https://coder.sqcorp.co/bundle.82f6fac05ecb071bdc27.js:2:385710)
    at _l (https://coder.sqcorp.co/bundle.82f6fac05ecb071bdc27.js:2:372984)
    at Cl (https://coder.sqcorp.co/bundle.82f6fac05ecb071bdc27.js:2:372912)
    at kl (https://coder.sqcorp.co/bundle.82f6fac05ecb071bdc27.js:2:372775)
    at vl (https://coder.sqcorp.co/bundle.82f6fac05ecb071bdc27.js:2:369762)
    at https://coder.sqcorp.co/bundle.82f6fac05ecb071bdc27.js:2:319583
    at t.unstable_runWithPriority (https://coder.sqcorp.co/bundle.82f6fac05ecb071bdc27.js:2:418114)
    at Vo (https://coder.sqcorp.co/bundle.82f6fac05ecb071bdc27.js:2:319360)

============================================================
```

I wasn't using the `aws-linux` example exactly but my own AWS Linux template that I was previously able to create a workspace from in the web UI until I added this `validation` block to one of the template variables:

```
validation {
  condition     = trimprefix(var.amazon_machine_image_name, "my-prefix-") != var.amazon_machine_image_name
  error_message = "The AMI name must start with \"my-prefix-\"."
}
```

Creating a workspace from the CLI would still work without issues, though.

Once I remove that `validation` block and re-upload my template, I am able to create a workspace from it in the Coder web UI once more.

**Comment by Kira-Pilot** [comment]
Tried this out this morning but couldn't reproduce 🤔

**Comment by Kira-Pilot** [comment]
This should be resolved by #2384.
